### ISSUE TITLE
Sidebar content is not displayed with Angular 5 on IE11

### ISSUE THREAD

**Issue by codech3ry** [arkon/ng-sidebar#131]
I'm using the latest version of ng-sidebar with Angular 5 (clean angular-cli project).
When the sidebar is initially displayed with opened=true, the sidebar is displayed fine but when I close it and reopen, **aside** tag has stale **style="transform: translateX(-100%)"** and hence the sidebar content is not displayed. 
Sidebar does open and close but its content remains hidden and doesn't move along with the sidebar.

_When the sidebar is open:_ 

`<aside class="demo-sidebar ng-sidebar--animate ng-sidebar ng-sidebar--opened ng-sidebar--left ng-sidebar--push" role="complementary" aria-hidden="false" style="transform: translateX(-100%);" aria-label="My sidebar" ng-reflect-klass="ng-sidebar ng-sidebar--opened " ng-reflect-ng-style="[object Object]" ng-reflect-ng-class="demo-sidebar" _ngcontent-c3="">
`

Might be related to #104 ?

"angular/animations": "^5.0.0",
"angular/common": "^5.0.0",
"angular/compiler": "^5.0.0",
"angular/core": "^5.0.0",
"angular/forms": "^5.0.0",
"angular/http": "^5.0.0",
"angular/platform-browser": "^5.0.0",
"angular/platform-browser-dynamic": "^5.0.0",
"angular/router": "^5.0.0",
"ng-sidebar": "^6.0.4",

**Comment by iagocanalejas** [comment]
Same for me with angular 5.0.3 in Chrome

**Comment by arkon** [comment]
Can't seem to reproduce it on my end with Angular 5.0.3 in Chrome. What settings do you have, @iagocanalejas ?

**Comment by iagocanalejas** [comment]
Hi, i'm using ng-sidebar 6.0.4 but have test all 6.0.x and the error is the same, my html looks like:
```html
<div class="container-fluid main-container-fluid">
  <ng-sidebar-container>
    <!-- A sidebar -->
    <ng-sidebar #sidebar [mode]="'over'" [closeOnClickBackdrop]="true" [showBackdrop]="true">
      <navbar [brandImage]="'../assets/img/exercise_placeholder.png'" [sidebar]="sidebar"></navbar>
    </ng-sidebar>

    <!-- NavBar -->
    <nav class="navbar navbar-expand-xl">
      ------
    </nav>

    <!-- Page Content -->
    <div ng-sidebar-content class="container-fluid text-center">
      <router-outlet></router-outlet>
    </div>
  </ng-sidebar-container>

</div>
```

Dependencies:
    "@angular/animations": "^5.0.3",
    "@angular/common": "^5.0.3",
    "@angular/compiler": "^5.0.3",
    "@angular/core": "^5.0.3",
    "@angular/forms": "^5.0.3",
    "@angular/http": "^5.0.3",
    "@angular/platform-browser": "^5.0.3",
    "@angular/platform-browser-dynamic": "^5.0.3",
    "@angular/router": "^5.0.3",
    "ng-sidebar": "^6.0.4",

**Comment by ChronSyn** [comment]
I also have the same issue. It appears that the `transform: translateX(100%);` isn't being removed correctly in IE (tested on IE11), yet the same problem isn't present in Chrome.

**Comment by sJohnsonStoever** [comment]
I'm having the same issue... noticed only in IE 11 (don't have access to 10) after making the jump to Angular 5.

ng-cli AOT (ng build --prod --aot --build-optimizer)

## Browsers

### On IE 11 (11.0.9600.18860):

When closed:

```
<aside class="ng-sidebar ng-sidebar--closed ng-sidebar--left ng-sidebar--slide main-sidebar-default ng-sidebar--docked ng-sidebar--animate" role="complementary" aria-hidden="true" style="transform: translateX(-100%) translateX(30px);" _ngcontent-c2="">
...
</aside>
```

When open:

```
<aside class="main-sidebar-default ng-sidebar--animate ng-sidebar ng-sidebar--opened ng-sidebar--left ng-sidebar--slide" role="complementary" aria-hidden="false" style="transform: translateX(-100%) translateX(30px);" _ngcontent-c2="">
...
</aside>
```

### On Firefox (52.5.2):

When closed:

```
<aside _ngcontent-c2="" role="complementary" class="ng-sidebar ng-sidebar--closed ng-sidebar--left ng-sidebar--slide main-sidebar-default ng-sidebar--docked ng-sidebar--animate" style="transform: translateX(-100%) translateX(30px);" aria-hidden="true">
...
</aside>
```

When open:

```
<aside _ngcontent-c2="" role="complementary" class="main-sidebar-default ng-sidebar--animate ng-sidebar ng-sidebar--opened ng-sidebar--left ng-sidebar--slide" style="" aria-hidden="false">
...
</aside>
```

### On Chrome (63.0.3239.84)

When closed:

```
<aside _ngcontent-c2="" role="complementary" class="main-sidebar-default ng-sidebar--animate ng-sidebar ng-sidebar--closed ng-sidebar--left ng-sidebar--slide ng-sidebar--docked" aria-hidden="true" style="transform: translateX(-100%) translateX(30px);">
...
</aside>
```

When open:
```
<aside _ngcontent-c2="" role="complementary" class="main-sidebar-default ng-sidebar--animate ng-sidebar ng-sidebar--opened ng-sidebar--left ng-sidebar--slide" aria-hidden="false" style="">
...
</aside>
```

## Temporary Fix

I've been able to build a custom version of ng-bootstrap for the time being that fixes this issue.

In file `sidebar-container.component.ts` function `_getContentStyle()`, change the initial values for the variables from `null` to and emtpy string:

```
let transformStyle: string = '';
let heightStyle: string = '';
let widthStyle: string = '';
```

Likewise for `_getStyle()` in `sidebar.component.ts`:

```
let transformStyle: string = '';
```

This fixes the issue, though it seems that the real bug is likely with IE and NgStyle in @angular/core, so I'm not sure if this is something that should be fixed in ng-sidebar or if an issue should be opened with Angular.

**Comment by arkon** [comment]
Thank you so much for investigating, @sJohnsonStoever . It does seem to be a bug in Angular, so I'll open an issue there later.

Here's a Plunker showing the same issue: https://plnkr.co/edit/kcCH48on1MT6xShkU9Va?p=preview

**Comment by gstercken** [comment]
@sJohnsonStoever: Thanks for the investigation and bugfix, this has been bugging us for a while. We have been able to resolve the IE/Edge problem locally with your fix.

@arkon: Even though it seems to be an Angular bug - any objections to fixing it as @sJohnsonStoever suggested by simply initializing the styles to empty strings instead of null?

**Comment by arkon** [comment]
I don't mind including the fix, but Angular is giving me grief again when I'm trying it locally...

![screen shot 2017-12-26 at 11 28 30](https://user-images.githubusercontent.com/4098258/34360579-efc86534-ea2f-11e7-91ad-e725cb08d40d.png)

**Comment by arkon** [comment]
Hmm, seems like an issue with Angular 5, so I built it with Angular 4 for now (since the previously published version was too).

I've published the workaround in v6.0.5, let me know if it works.

**Comment by gstercken** [comment]
Interesting... I built locally with Angular 5.0.1 and got no issues, but I'll try again with the latest stable (5.1.2). Thanks for the fix! We'll try v6.0.5 and let you know.

**Comment by arkon** [comment]
Seems that it just got fixed upstream.
